### DISCUSSION TITLE
Why is BMP image height & width capped at 65535 pixels?

### DISCUSSION THREAD

**Discussion by smorokin** [SixLabors/ImageSharp#2191] [Q&A] [answered]
Hello,
is there a reason why when reading the BMP header the image [width and height checked to be less then 65536 pixels __each__](https://github.com/SixLabors/ImageSharp/blame/main/src/ImageSharp/Formats/Bmp/BmpInfoHeader.cs#L548). 
Wikipedia claims that BMP image width or height could be up to 2<sup>31</sup>-1 each, but as i understand it their product must also be less then 2<sup>32</sup>, so I assume thats where the 65535 limit comes from (sqrt(2<sup>32</sup>) = 65535). 
I would like to propose to change the check to height and width <2<sup>31</sup> each _and_ height * width < 2<sup>32</sup>, but wanted to know if there are some other limitations that I have not found, first.

**Comment by JimBobSquarePants** [comment, marked as answer]
It's because early bitmap versions only allowed 2 bytes max per dimension. We should actually update the rules to max out based upon the header version. Happy to accept a PR if you're keen.

**Reply by smorokin** [reply]
I have created a PR draft #2192 . Is a test neccessary?

**Reply by smorokin** [reply]
Oh, sorry, I have missed the "based upon the header version". I will have another look
